exporter/elasticexporter/config.go

```diff
@@ -16,16 +16,25 @@ package elasticexporter
 
 import (
 	"errors"
+	"fmt"
+	"strings"
+	"time"
 
 	"go.opentelemetry.io/collector/config/configmodels"
 	"go.opentelemetry.io/collector/config/configtls"
 )
 
-// Config defines configuration for Elastic APM exporter.
+// Config defines configuration for Elastic exporter.
 type Config struct {
 	configmodels.ExporterSettings `mapstructure:",squash"`
 	configtls.TLSClientSetting    `mapstructure:",squash"`
 
+	APMServerConfig     `mapstructure:",squash"`
+	ElasticsearchConfig `mapstructure:",squash"`
```

Comment:
I'm a bit confused by having both apm server and elasticsearch here. It seems like they are quite different APIs - since we already have translation in an internal shared package anyways, does it make sense to split into two packages, `elasticapmexporter` and `elasticsearchexporter`?

Reply:
@anuraaga this decision to have one single exporter whose implementation pushes data to two different endpoints is the result of a conversation with @tigrannajaryan (see https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/1630#issuecomment-729894364 ).
We changed our point of view and subscribe to Tigran's point of view, we feel this will be more straightforward for users.

Reply:
The "one exporter for multiple signals" is a recommendation. That's what OTLP exporter does. In many cases it results in less config settings for the end user to specify (e.g. you likely need to specify access tokens and other credentials, so you do it once, in one exporter config section).
It is not a hard rule, feel free to deviate if you feel there is not much common between the exporters of different signals.

Reply:
> multiple signals

My understanding of this is that this isn't quite multiple signals but two different backends, elasticsearch and elastic apm. While logs are here only exported to elasticsearch I guess we can assume elastic apm would also ingest them at some point? And maybe traces would be exported to Elasticsearch similar to Zipkin's support.

So if it's two backends I guess it's natural to have to exporter packages.

Reply:
Thanks for the feedback @tigrannajaryan and @anuraaga .
We liked Tigran's suggestion and we got positive feedback from users on the UX we would have, it was perceived as pretty intuitive.

> So if it's two backends I guess it's natural to have to exporter packages.

We would like to get the feedback of few users on the User Experience with two exporters, we should be able to share our findings very soon on this thread.
In parallel, we continue the implementation.

Reply:
Hello, sorry for the delay. We have confirmed with a bunch of users that the user experience is intuitive enough with a split of exporters, introducing an `elasticsearch` exporter.

We will move forward creating a PR for the new `Elasticsearch`exporter and we will close this PR. Does it make sense @anuraaga and @tigrannajaryan ?

Reply:
If you feel that works best for you then it works for me.

Reply:
Thank you @tigrannajaryan